File: 16-array-sum.rb
# get the maximum number from an array

# def find_max(array)
#   array.sort.last
# end

def find_max(array)
  n = array[0]
  array.each do |elem|
    if elem > n
      n = elem
    end
  end
  n
end

arr = [8, 12, 36, 53, 9, 75, 3, 71, 59, 88]

max = find_max(arr)
puts "Max is #{max}" # should be 88

File: 16-array-sum/16-array-sum.rs
// 给定一阵列内含数字，输出最大值

fn find_max(arr: &Vec<i32>) -> &i32 {
    arr.iter().max().unwrap()
}

fn main() {
    let arr: Vec<i32> = vec![8, 12, 36, 53, 9, 75, 3, 71, 59, 88];

    let max = find_max(&arr);
    println!("Max is {max}"); // 应该是 88
}
